zachsinterview/src/main/java/com/imprev/zachsinterview/NumberStoreImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.imprev.zachsinterview;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+import com.imprev.zachsinterview.thirdpartyapi.ThirdPartyDictionaryAPI.InvalidCredentialsException;
+import com.imprev.zachsinterview.thirdpartyapi.ThirdPartyDictionaryAPI.TryAgainLaterException;
+import com.imprev.zachsinterview.thirdpartyapi.ThirdPartyDictionaryAPIImpl;
+import com.imprev.zachsinterview.thirdpartyapi.ThirdPartyDictionaryAPI;
+
+/**
+ * This class wraps the httpClient that will communicate with our partners third party dictionary store.  The third party api
+ * stores values per key as a list of strings, so this will manage the transformation to and from an integer value
+ * to a list of strings.  Additionally this class will add some retry logic in the event we encounter an exception 
+ * with the third party API;
+ *
+ */
+public class NumberStoreImpl implements NumberStore {
+
+    private ThirdPartyDictionaryAPI thirdPartyDictionaryAPIClient;
+
+    private int MAX_RETRIES = 3;
+
+    public NumberStoreImpl (
+            String username,
+            String secretPassword,
+            String host,
+            int port) {
+
+        this.thirdPartyDictionaryAPIClient = new ThirdPartyDictionaryAPIImpl(username, secretPassword, host, port);
+
+        // lets ping the partner to verify our credentials work!
+        try {
+            this.thirdPartyDictionaryAPIClient.verifyCredentials();
```

Comment:
would be best to retry verifyCredentials/initiating the connection in case the first attempt fails

Reply:
would be best to verifyCredentials at each API call in case the credentials expire, and re-request credentials as needed, allowing work to continue without interruption.